### PR TITLE
chore(node): Add edge runtime test coverage

### PR DESCRIPTION
**Pull request by dustinbyrne**
## Problem

The Node SDK exports an edge client, but we do not regularly validate tests in an edge runtime.

## Changes

- Add `test:edge-compat` script to run Node tests under `@edge-runtime/jest-environment`
- Add edge-compat tests to CI                                        
- Replace `AbortSignal.timeout()` in `fetchWithRetry` with manual `AbortController` + `clearTimeout` because the old approach leaked timers in edge runtimes where `setTimeout` returns numeric handles that can't be `unref()`'d
- Same fix for `_shutdown`'s `Promise.race` timeout
- Add test verifying the AbortController abort fires on slow requests

There is a patch release included in this PR but it's mostly cleanup to ensure we don't leak timers.

### Libraries affected

- [ ] All of them
- [ ] posthog-js (web)
- [ ] posthog-js-lite (web lite)
- [x] posthog-node
- [ ] posthog-react-native
- [ ] @posthog/react
- [ ] @posthog/ai
- [ ] @posthog/convex
- [x] @posthog/core
- [ ] @posthog/next
- [ ] @posthog/nextjs-config
- [ ] @posthog/nuxt
- [ ] @posthog/rollup-plugin
- [ ] @posthog/webpack-plugin
- [ ] @posthog/types

## Checklist

- [x] Tests for new code
- [x] Accounted for the impact of any changes across different platforms
- [x] Accounted for backwards compatibility of any changes (no breaking changes!)
- [x] Took care not to unnecessarily increase the bundle size

### If releasing new changes

- [x] Ran `pnpm changeset` to generate a changeset file
- [x] Added the "release" label to the PR to indicate we're publishing new versions for the affected packages

### CODE REVIEW

File: .changeset/good-ties-double.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@posthog/core': patch
```
Comment (marandaneto) [inline .changeset/good-ties-double.md:2]: since node depends on @posthog/core, dont we need a node patch as well?

Comment (dustinbyrne) [reply .changeset/good-ties-double.md:2]: I believe it will automatically patch bump dependencies - will double check
